test/test-lex.cpp

```diff
@@ -196,13 +327,6 @@ TEST(test_lex, lex_invalid_big_int_number) {
                                 error_big_int_literal_contains_decimal_point,
                                 where, offsets_matcher(input, 0, 6))));
       });
-  check_tokens_with_errors(
```

Comment:
This change implies that the `number_begin[0] == u8'0'` check in `lexer::parse_number` will always be false. Clean up that dead code?